src/plugins/encryption/Cipher.h

```diff
@@ -0,0 +1,23 @@
+//
+// Created by moorts on 22.10.21.
+//
+#include <string>
+#include <cryptopp/modes.h>
+#include <cryptopp/osrng.h>
+
+#include <iostream>
+
+#ifndef SRC_CIPHER_H
+#define SRC_CIPHER_H
+
+using namespace CryptoPP;
+
+class Cipher {
```

Comment:
This class should be moved out of plugins and into the application layer

Reply:
I'm not so sure thats a good idea, since the implementations of Cipher directly call the libraries functions

Reply:
Yes exactly, the implementations do, but this class doesn't. This class is used to inject the dependency into the EncryptionInterface, which correctly noticed by you as well, should be in the application layer. Therefore if this class isn't in the application layer there would be a dependency pointing from the application layer to the plugins layer. However, dependencies should only go from the outside towards the inside and not the other way around.